packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -265,21 +261,26 @@ class DataSourceWithBackend<
   }
 
   /**
-   * Apply template variables for explore
+   * Filters out queries that are empty or hidden. Used when running queries through backend.
+   * @returns `true` if the query is not hidden and its expression is not empty; `false` otherwise.
    */
-  interpolateVariablesInQueries(queries: TQuery[], scopedVars: ScopedVars, filters?: AdHocVariableFilter[]): TQuery[] {
-    return queries.map((q) => this.applyTemplateVariables(q, scopedVars, filters) as TQuery);
+  filterQuery(query: TQuery): boolean {
+    if (query.hide) {
+      return false;
+    }
+    if (super.isCompleteQuery) {
+      return super.isCompleteQuery(query);
```

Comment:
Have you tested this? I don't think it will work. `isCompleteQuery` is defined in the abstract base class, so even though a data source implements this method it will never be called.

Reply:
I hadn't tested it, I was trying to float some ideas to see what would work best with what people were requesting. I'll incorporate this into a proposal in a design doc which will be more clear than trying to figure it out in a PR.